pixeltable/catalog/catalog.py

```diff
@@ -98,13 +97,12 @@ def loop(*args: Any, **kwargs: Any) -> T:
                         for_write=for_write,
                         convert_db_excs=False,
                         lock_mutable_tree=lock_mutable_tree,
-                        finalize_pending_ops=finalize_pending_ops,
+                        finalize_pending_ops=True,
                     ):
                         return op(*args, **kwargs)
                 except PendingTableOpsError as e:
-                    assert finalize_pending_ops
                     _logger.debug(f'retry_loop(): finalizing pending ops for {e.tbl_id}')
```

Comment:
If you are using the console logger there is no reason to also put a separate `_logger.debug()` (this will just result in duplicate messages in the logs)

Same comment applies to other uses of console logger below.

Reply:
Done.